packages/create-next-app/index.ts

```diff
@@ -137,307 +111,358 @@ const program = new Commander.Command(packageJson.name)
+    opts.importAlias = undefined
+  }
+
+  if (!opts.importAlias && !args.includes('--no-import-alias')) {
+    const { customizeImportAlias } = await _prompt({
+      type: 'toggle',
+      name: 'customizeImportAlias',
+      message: `Would you like to customize the default ${ct.importAlias} (${defaults.importAlias})?`,
+    })
+
+    if (customizeImportAlias) {
+      const { importAlias } = await _prompt({
+        type: 'text',
+        name: 'importAlias',
+        message: `How would you like to configure the ${ct.importAlias}?`,
+        validate: (value) => validateImportAlias(value),
+      })
+      opts.importAlias = importAlias
+      preferences.importAlias = importAlias
     }
+  }
 
-    const importAliasPattern = /^[^*"]+\/\*\s*$/
-    if (
-      typeof program.importAlias !== 'string' ||
-      !importAliasPattern.test(program.importAlias)
-    ) {
-      if (ciInfo.isCI) {
-        // We don't use preferences here because the default value is @/* regardless of existing preferences
-        program.importAlias = defaults.importAlias
-      } else if (process.argv.includes('--no-import-alias')) {
-        program.importAlias = defaults.importAlias
-      } else {
-        const styledImportAlias = blue('import alias')
-
-        const { customizeImportAlias } = await prompts({
-          onState: onPromptState,
-          type: 'toggle',
-          name: 'customizeImportAlias',
-          message: `Would you like to customize the default ${styledImportAlias} (${defaults.importAlias})?`,
-          initial: getPrefOrDefault('customizeImportAlias'),
-          active: 'Yes',
-          inactive: 'No',
-        })
-
-        if (!customizeImportAlias) {
-          // We don't use preferences here because the default value is @/* regardless of existing preferences
-          program.importAlias = defaults.importAlias
-        } else {
-          const { importAlias } = await prompts({
-            onState: onPromptState,
-            type: 'text',
-            name: 'importAlias',
-            message: `What ${styledImportAlias} would you like configured?`,
-            initial: getPrefOrDefault('importAlias'),
-            validate: (value) =>
-              importAliasPattern.test(value)
-                ? true
-                : 'Import alias must follow the pattern <prefix>/*',
-          })
-          program.importAlias = importAlias
-          preferences.importAlias = importAlias
-        }
-      }
+  // Ensure the importAlias is set.
+  opts.importAlias ??= defaults.importAlias
+
+  const resolvedOpts: ResolvedCreateNextAppOptions = {
+    typescript: Boolean(opts.typescript),
+    eslint: Boolean(opts.eslint),
+    tailwind: Boolean(opts.tailwind),
+    app: Boolean(opts.app),
+    srcDir: Boolean(opts.srcDir),
+    importAlias: opts.importAlias,
+  }
+  await tryCreateNextApp({
+    appPath,
+    resolvedOpts,
+    conf,
+    preferences,
+    isDryRun,
+  })
+}
+
+async function tryCreateNextApp({
```

Comment:
Split create app tasks as a separate function and run them based on the conditions: `example`, `isCI`, and basic.